code/modules/antagonists/wizard/equipment/soulstone.dm

```diff
@@ -69,7 +69,7 @@
 
 //////////////////////////////Capturing////////////////////////////////////////////////////////
 
-/obj/item/soulstone/attack(mob/living/carbon/human/M, mob/living/user)
+/obj/item/soulstone/attack(mob/living/carbon/human/M, mob/living/user, key)
```

Comment:
I am pretty sure attack() proc doesn't pass key separately and you did something pretty much useless.
Gonna add a code suggestion below.